pkg/lwip/contrib/sock/udp/lwip_sock_udp.c

```diff
@@ -120,23 +120,22 @@ ssize_t sock_udp_recv_buf(sock_udp_t *sock, void **data, void **ctx,
     if (remote != NULL) {
         /* convert remote */
         size_t addr_len;
-#if LWIP_IPV6
-        if (sock->base.conn->type & NETCONN_TYPE_IPV6) {
-            addr_len = sizeof(ipv6_addr_t);
-            remote->family = AF_INET6;
+        if (IS_USED(LWIP_IPV6)) {
+            if (!IS_USED(LWIP_IPV4) ||
```

Comment:
This is a logical change to what was here before. Please explain, as I don't really understand the reasoning.

Reply:
If IPv6 and only IPv6 is used, the address family will be IPv6. (In the IPv4 only case it already was setting the family to IPv4 without check, I just applied the same optimization to both single-stack cases.)